src/java/org/apache/cassandra/cql3/functions/CastFcts.java

```diff
@@ -126,25 +130,38 @@ private static <I extends Number> java.util.function.Function<I, BigDecimal> get
     }
 
     /**
-     * Creates the name of the cast function use to cast to the specified type.
+     * Creates the snake-cased name of the cast function used to cast to the specified type.
      *
      * @param outputType the output type
-     * @return the name of the cast function use to cast to the specified type
+     * @return the name of the cast function used to cast to the specified type
      */
-    public static String getFunctionName(AbstractType<?> outputType)
+    public static String getFunctionName(CQL3Type outputType)
     {
-        return getFunctionName(outputType.asCQL3Type());
+        return FUNCTION_NAME_PREFIX + toLowerCaseString(outputType);
     }
 
     /**
-     * Creates the name of the cast function use to cast to the specified type.
+     * Creates the legacy camel-cased name of the cast function used to cast to the specified type.
      *
      * @param outputType the output type
-     * @return the name of the cast function use to cast to the specified type
+     * @return the legacy camel-cased name of the cast function used to cast to the specified type
      */
-    public static String getFunctionName(CQL3Type outputType)
+    private static String getLegacyFunctionName(CQL3Type outputType)
+    {
+        return LEGACY_FUNCTION_NAME_PREFIX + WordUtils.capitalize(toLowerCaseString(outputType));
```

Comment:
Will `WordUtils.capitalize()` ever kick in? It's going to be fed the lowercase class name, iiuc, which will never have blank spaces

Reply:
It would, since the first word be capitalized independently of spaces or underscores. For example, `int` would become `Int`.

What we could do here, by the way, is replacing the deprecated `WordUtils.capitalize` by the newer `StringUtils.capitalize`.